### PR TITLE
wip, vector{P} proper

### PR DESCRIPTION
**Pull request by dehann**

### CODE REVIEW

File: src/Interface.jl [hunk 1]
```diff
@@ -44,19 +47,49 @@ function marginal(x::ManifoldKernelDensity{M,B},
...
+
+Notes
+- `u0` is used to identify the data type for a point
+- Pass in a different `exp` if needed.
+"""
+makeManifoldPoint(M::MB.AbstractManifold,
+                  coords::AbstractVector{<:Real},
+                  u0=zeros(0),
+                  ϵ=identity(M,u0),
+                  _exp::Function=exp ) = _exp(M, ϵ, hat(M, ϵ, coords))
```
Comment (Affie) [inline src/Interface.jl:77]: Just as reference, it look like a more general way to do this is to use:
`retraction_method::AbstractRetractionMethod=ExponentialRetraction()`
`retract(M, p, X, retraction_method)`

File: src/ApproxManifoldProducts.jl [hunk 2]
```diff
@@ -1,5 +1,6 @@
 module ApproxManifoldProducts
 
+using Base: print_stackframe
```
Comment (Affie) [inline src/ApproxManifoldProducts.jl:3]: I think this slipped in by mistake?

**Comment by Affie** [comment]
I'll give it a test once you are done

File: src/ApproxManifoldProducts.jl (hunk 2 above)
Comment (dehann) [reply src/ApproxManifoldProducts.jl:3]: huh, not sure what that is.  Will remove thanks

File: src/Interface.jl (hunk 1 above)
Comment (dehann) [reply src/Interface.jl:77]: retract it is!

**Comment by dehann** [comment]
Hi @Affie , all right, tests here on AMP are passing again.  You welcome to give it a try.  More cleanup of legacy functions and several function duplications.   I tried to get similar functions in the same files with this PR.

**Pull request merged**